src/bp/core/routers/bots/index.ts

```diff
@@ -281,15 +281,52 @@ export class BotsRouter extends CustomRouter {
     )
 
     this.router.post(
-      '/flows',
+      '/flow',
       this.checkTokenHeader,
       this.needPermissions('write', 'bot.flows'),
       this.asyncMiddleware(async (req, res) => {
         const botId = req.params.botId
-        const flowViews = <FlowView[]>req.body.dirtyFlows
+        const flow = <FlowView>req.body.flow
 
-        await this.flowService.saveAll(botId, flowViews, req.body.cleanFlows)
-        res.sendStatus(201)
+        await this.flowService.upsertFlow(botId, flow)
+        res.sendStatus(200)
+      })
+    )
+
+    this.router.put(
+      '/flow/:flowName',
+      this.checkTokenHeader,
+      this.needPermissions('write', 'bot.flows'),
+      this.asyncMiddleware(async (req, res) => {
+        const botId = req.params.botId
+        const flowName = req.params.flowName
+        const flow = <FlowView>req.body.flow
+
+        if (_.has(flow, 'name') && flowName !== flow.name) {
```

Comment:
Usually we use lodash to check a complex path (eg: flow, 'some.deep.property'). Here, it would be clearer using
if(flow.name && flow.name !== flowName

Reply:
Here it stands for:

```
if(flow && flow.name && flow.name !== flowName) {}
```

which is kind of long.

_.has has the advantage of asserting both flow and flow.name are defined